android/src/main/java/com/amplitude/android/Configuration.kt

```diff
@@ -40,6 +40,9 @@ open class Configuration @JvmOverloads constructor(
     var flushEventsOnClose: Boolean = true,
     var minTimeBetweenSessionsMillis: Long = MIN_TIME_BETWEEN_SESSIONS_MILLIS,
     var trackingSessionEvents: Boolean = true,
+    var trackingAppLifecycleEvents: Boolean = false,
+    var trackingDeepLinks: Boolean = false,
+    var trackingScreenViews: Boolean = false,
```

Comment:
cc @kevinpagtakhan

Reply:
yes please! and if we can do the `boolean | Options` patter

Reply:
Thanks @justin-fiedler @kevinpagtakhan! I thought about this, my concerns about the union type are:
- The `trackingSessionEvents` already exists, if we introduce a new `defaultTracking` option with `Boolean` or `Options`, it might introduce some confusion or backward-compatible issues. Plus the session events are enabled by default, while other events are disabled by default.
- Kotlin doesn't support union type by default (https://youtrack.jetbrains.com/issue/KT-13108/Denotable-union-and-intersection-types, currently under discussion whether to support). As far as I have searched, if we want to achieve this, then we would need to expand the type to `Any` (see [example](https://pl.kotl.in/WOeBC8p3C)), or probably introduce some library like [`Either`](https://apidocs.arrow-kt.io/arrow-core/arrow.core/-either/index.html) in Arrow (haven't tried it, not sure whether it is IDE friendly or not).

Let me know your thoughts!

Reply:
Thanks @liuyang1520. Good points

- "`trackingSessionEvents` already exists" - I think we could mark it as deprecated and give precedence to `defaultTracking` configuration in the case `defaultTracking` is provided by the user.
- I don't think we necessarily need Union types. Instead we could maybe expose some common defaults as static objects to make it easy set a single value. Example (rough) pseudocode

```
open class AttributionOptions @JvmOverloads constructor(
  var excludeReferrers, ...
) {
  companion object {
     const val All = AttributionOptions(...defaultsForAll)
  }
}
class DefaultTrackingOptions @JvmOverloads constructor(
  var attributionOptions: AttributionOptions, ...
) {
  companion object {
     const val All = DefaultTrackingOptions(...defaultsForAll)
     const val None = DefaultTrackingOptions(...defaultsForNone)
  }
}

Amplitude(Configuration(defaultTracking: DefaultTrackingOptions.ALL))
```

Reply:
i agree with justin on how to move forward with `trackingSessionEvents`. i think the nested config is the "new" way now and would prefer that we have more "new" ways than more "old" ways. sets us up better for 2.x

Reply:
Thanks @justin-fiedler , @kevinpagtakhan! Good idea on the static objects, I will add the deprecate tag to `trackingSessionEvents` and add some common defaults as static objects.